test/unit/test_discovery_v1.rb

```diff
@@ -1010,6 +1010,171 @@ def test_list_training_examples
     assert_equal({ "received" => "true" }, service_response.result)
   end
 
+  def test_create_event
+    data_create_event = {
+      environment_id: "envid",
+      session_token: "token",
+      collection_id: "collid",
+      document_id: "docid"
+    }
+    mock_response = {
+      "type" => "click",
+      "data" => "data"
+    }
+    service = IBMWatson::DiscoveryV1.new(
+      username: "username",
+      password: "password",
+      version: "2018-05-23"
+    )
+    stub_request(:post, "https://gateway.watsonplatform.net/discovery/api/v1/events?version=2018-05-23")
+      .with(
+        body: "{\"type\":\"click\",\"data\":{\"environment_id\":\"envid\",\"session_token\":\"token\",\"collection_id\":\"collid\",\"document_id\":\"docid\"}}",
+        headers: {
+          "Accept" => "application/json",
+          "Authorization" => "Basic dXNlcm5hbWU6cGFzc3dvcmQ=",
+          "Content-Type" => "application/json",
+          "Host" => "gateway.watsonplatform.net"
+        }
+      )
+      .to_return(status: 201, body: mock_response.to_json, headers: {})
```

Comment:
While I'm sure the `test_create_event` test passes, one of the headers in the sent request is `"Accept" => "application/json"`, which means that the method is expecting to receive a response with its data in an `application/json` format and a response header with `"Content-Type" => "application/json"`. The assert should be comparing two hashes, but is instead comparing two strings. I don't know if this is me just nitpicking or whatever. Basically, to fix this (I think), add a `"Content-Type" => "application/json"` to the `to_return` headers, and remove the `.to_json` from the `assert_equal`.

Reply:
Good catch!